test/unit/lib/update-lock.spec.ts

```diff
@@ -14,4 +17,52 @@ describe('lib/update-lock: unit tests', () => {
 			);
 		});
 	});
+
+	describe('getServicesLockedByAppId', () => {
+		let getLocksTaken: SinonStub;
+		beforeEach(() => {
+			getLocksTaken = stub(lockfile, 'getLocksTaken');
```

Comment:
I don't like this stubbing of internal methods, I think we should avoid it from now on as it means the test is relying on internal implementation details of the function being tested. Either the environment should be setup in advance to what the function expects (you can use mocha-pod to help with that), or the function should receive `locksTaken` as argument and make it a pure function.

Reply:
Noted, will change

Reply:
Removed the stubs in favor of testFs [here](https://github.com/balena-os/balena-supervisor/pull/2234/commits/b5d5dd7a9837a8a939cea5d99fddc56f2d4840ac#diff-e7a2b32a616114c913967bbeb93301d1194c25d37b5c6d6393146a862d33b61fR285-R347). I wasn't able to get testfs empty directories to work consistently. I'd like to look at it more closely, but perhaps in a separate PR.